### PR TITLE
STACKIT Kubernetes API load balancing

### PR DESCRIPTION
**Pull request by 3u13r**
### Proposed change(s)

- Add stackit load balancer in OpenStack Terraform
- Get LoadBalancerEndpoint in OpenStack from the IMDS API

Note that once we want to support more openstack implementation than StackIT, then we should make StackIT its own provider.

To test this PR you need to be authenticated with the StackIT API. You can follow one of the ways described here: https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs#authentication

### Additional info

- [AB#3882](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3882)

### Checklist

- [ ] Run the E2E tests that are relevant to this PR's changes
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: cli/internal/cloudcmd/tfvars.go [hunk 1]
```diff
@@ -266,6 +266,7 @@ func openStackTerraformVars(conf *config.Config, imageRef string) (*terraform.Op
 		NodeGroups:              nodeGroups,
 		CustomEndpoint:          conf.CustomEndpoint,
 		InternalLoadBalancer:    conf.InternalLoadBalancer,
+		StackITProjectID:        conf.Provider.OpenStack.StackITProjectID,
```
Comment (malt3) [inline cli/internal/cloudcmd/tfvars.go:269]:
```suggestion
		STACKITProjectID:        conf.Provider.OpenStack.STACKITProjectID,
```

File: cli/internal/terraform/variables.go [hunk 2]
```diff
@@ -280,6 +280,8 @@ type OpenStackClusterVariables struct {
 	NodeGroups map[string]OpenStackNodeGroup `hcl:"node_groups" cty:"node_groups"`
 	// Cloud is the (optional) name of the OpenStack cloud to use when reading the "clouds.yaml" configuration file. If empty, environment variables are used.
 	Cloud *string `hcl:"cloud" cty:"cloud"`
+	// (StackIT only) StackITProjectID is the ID of the StackIT project to use.
+	StackITProjectID string `hcl:"stackit_project_id" cty:"stackit_project_id"`
```
Comment (malt3) [inline cli/internal/terraform/variables.go:284]:
```suggestion
	// (STACKIT only) STACKITProjectID is the ID of the STACKIT project to use.
	STACKITProjectID string `hcl:"stackit_project_id" cty:"stackit_project_id"`
```

File: cli/internal/terraform/variables_test.go [hunk 3]
```diff
@@ -260,6 +260,7 @@ func TestOpenStackClusterVariables(t *testing.T) {
 		OpenstackUsername:       "my-username",
 		OpenstackPassword:       "my-password",
 		Debug:                   true,
+		StackITProjectID:        "my-stackit-project-id",
```
Comment (malt3) [inline cli/internal/terraform/variables_test.go:263]:
```suggestion
		STACKITProjectID:        "my-stackit-project-id",
```

File: internal/config/config.go [hunk 4]
```diff
@@ -207,9 +207,13 @@ type OpenStackConfig struct {
 	// AuthURL is the OpenStack Identity endpoint to use inside the cluster.
 	AuthURL string `yaml:"authURL" validate:"required"`
 	// description: |
-	//   ProjectID is the ID of the project where a user resides.
+	//   ProjectID is the ID of the OpenStack project where a user resides.
 	ProjectID string `yaml:"projectID" validate:"required"`
 	// description: |
+	//   StackITProjectID is the ID of the StackIT project where a user resides.
+	//   Only used if cloud is "stackit".
+	StackITProjectID string `yaml:"stackITProjectID"`
```
Comment (malt3) [inline internal/config/config.go:215]:
```suggestion
	//   STACKITProjectID is the ID of the STACKIT project where a user resides.
	//   Only used if cloud is "stackit".
	STACKITProjectID string `yaml:"stackitProjectID"`
```

File: terraform/infrastructure/openstack/variables.tf [hunk 5]
```diff
@@ -67,3 +67,10 @@ variable "openstack_password" {
   type        = string
   description = "OpenStack password."
 }
+
+# StackIT-specific variables
```
Comment (malt3) [inline terraform/infrastructure/openstack/variables.tf:71]:
```suggestion
# STACKIT-specific variables
```

File: internal/constellation/helm/overrides.go [hunk 6]
```diff
@@ -44,7 +44,9 @@ func extraCiliumValues(provider cloudprovider.Provider, conformanceMode bool, ou
 	}
 
 	strictMode := map[string]any{}
-	if provider != cloudprovider.QEMU {
+	// TODO(@3u13r): Once we are able to the subnet of the load balancer VMs
+	// on StackIT, we can remove the OpenStack exception here.
```
Comment (daniel-weisse) [inline internal/constellation/helm/overrides.go:48]: This comment seems incomplete

**Review by malt3 — COMMENTED** [review]
Code LGTM. Nice work! I'll perform a final test once we have this rebased on #2910.

File: terraform/infrastructure/openstack/main.tf [hunk 7]
```diff
@@ -26,14 +36,17 @@ locals {
...
-  ports_verify           = "30081"
-  ports_recovery         = "9999"
-  ports_debugd           = "4000"
-  cidr_vpc_subnet_nodes  = "192.168.178.0/24"
-  tags                   = ["constellation-uid-${local.uid}"]
+  control_plane_named_ports = flatten([
+    { name = "kubernetes", port = "6443", health_check = "HTTPS" },
+    { name = "bootstrapper", port = "9000", health_check = "TCP" },
+    { name = "verify", port = "30081", health_check = "TCP" },
+    { name = "konnectivity", port = "8132", health_check = "TCP" },
```
Comment (daniel-weisse) [inline terraform/infrastructure/openstack/main.tf:43]: We no longer deploy konnectivity, do we?
So this could be removed

**Review by daniel-weisse — APPROVED** [review]
Can't really comment on the Terraform changes.
Updates to my owned file look good to me

File: terraform/infrastructure/openstack/modules/stackit_loadbalancer/main.tf [hunk 8]
```diff
@@ -0,0 +1,47 @@
+terraform {
+  required_providers {
+    stackit = {
+      source  = "stackitcloud/stackit"
+      version = "0.12.0"
+    }
+  }
+}
+
+resource "stackit_loadbalancer" "example" {
```
Comment (malt3) [inline terraform/infrastructure/openstack/modules/stackit_loadbalancer/main.tf:10]:
```suggestion
resource "stackit_loadbalancer" "loadbalancer" {
```

File: terraform/infrastructure/openstack/main.tf [hunk 9]
```diff
@@ -176,44 +188,38 @@ resource "openstack_networking_secgroup_rule_v2" "nodeport_udp" {
...
 resource "openstack_networking_secgroup_rule_v2" "tcp_port_forward" {
-  for_each = toset(flatten([
-    local.ports_kubernetes,
-    local.ports_bootstrapper,
-    local.ports_konnectivity,
-    local.ports_verify,
-    local.ports_recovery,
-    var.debug ? [local.ports_debugd] : [],
-  ]))
+  for_each          = local.control_plane_named_ports
```
Comment (malt3) [inline terraform/infrastructure/openstack/main.tf:191]:
```suggestion
  for_each          = {for i,v in local.control_plane_named_ports: i=>v}
```

In my test, this was required

**Review by malt3 — APPROVED** [review]
Tested this successfully.

**Pull request merged**